dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESContentletAPIImpl.java

```diff
@@ -4397,6 +4404,42 @@ else if (incomingFile.exists() ){
         return contentlet;
     }
 
+    @VisibleForTesting
+    ESIndexAPI getEsIndexAPI() {
+        return esIndexAPI;
+    }
+
+    @VisibleForTesting
+    void setEsIndexAPI(ESIndexAPI esIndexAPI) {
+        this.esIndexAPI = esIndexAPI;
+    }
+
+    /**
+     * Method that verifies if a check in operation can be executed.
+     * It is safe to execute a checkin if write operations can be performed on the ES cluster.
+     * Otherwise, check in will be allowed only if the contentlet to be saved does not have legacy relationships
+     * @param relationships ContentletRelationships with the records to be saved
+     * @return
+     */
+    @VisibleForTesting
+    boolean isCheckInSafe(final ContentletRelationships relationships, final ESIndexAPI esIndexAPI) {
+
+        if (relationships != null && relationships.getRelationshipsRecords().size() > 0) {
+            boolean isClusterReadOnly = esIndexAPI.isClusterInReadOnlyMode();
+
+            if (isClusterReadOnly) {
+                for (ContentletRelationships.ContentletRelationshipRecords records : relationships
```

Comment:
maybe this code should go to a new method

```suggestion
boolean hasLegacyRelationships(...) {
                for (ContentletRelationships.ContentletRelationshipRecords records : relationships
                                              .getRelationshipsRecords()) {
                    if (!records.getRelationship().isRelationshipField()){
                        return true;
                    }
                }
                
                return false;
}                
```